include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -123,38 +125,58 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
 
     // TODO: filter hits and only do it when necessary (depending on error types)
 
+    constexpr bool output_cursor = cfg_t::template exists<search_cfg::output<detail::search_output_index_cursor>>();
+
+    using result_t = std::conditional_t<output_cursor,
+                                        search_result_value_type<typename index_t::cursor_type>,
+                                        search_result_value_type<typename index_t::cursor_type, size_t, size_t>>;
+
+    std::vector<result_t> results;
+
     // output cursors or text_positions
-    if constexpr (cfg_t::template exists<search_cfg::output<detail::search_output_index_cursor>>())
+    if constexpr (output_cursor)
     {
-        return internal_hits;
+        for (auto && cursor : internal_hits)
```

Comment:
so the hits are already generated? And here we copy them again? Or can the internal representation of internal hits already changed? Would that make sense?

Reply:
We have the cursors and need to move them into the `search_result_value_type`.... doesn't this move? `internal_hits` will go out of scope after this and won't change anymore.